backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java

```diff
@@ -570,7 +570,8 @@ public void onEvent(BacktraceResult backtraceResult) {
                 if (record != null) {
                     record.close();
                 }
-                if (backtraceResult != null && backtraceResult.status == BacktraceResultStatus.Ok) {
+
+                if (backtraceResult != null && record != null && backtraceResult.status == BacktraceResultStatus.Ok) {
                     database.delete(record);
```

Comment:
I think the record check should be in the database code rather than in the client. If we want to have such a check, can we move it to the database instead?

Reply:
it's extra check otherwise we may have weird nulls inside database context

Reply:
Yeah but it would be better if we put it in the database context entry point instead

Reply:
I've added extra null check in BacktraceDatabase, now we have it there, here and in BacktraceDatabaseContext